eclipse-theia/theia
# Reduce Preference Validation Warnings & Improve Monaco Extractor

#### What it does


This PR reduces the number of warnings logged by the `PreferenceValidationService` (hopefully to 0, unless real problems are created by user entry) by improving preference extraction from Monaco and adjusting a few schemata to reflect additional values expected by Monaco but not explicitly specified in the schema.

#### How to test


1. Open a workspace. Remove any custom preference settings.
2. Open an editor.
3. Observe that no warnings are logged by the preference validation system about preference values differing from the schema.
4. Set a preference to a disallowed value (e.g. a editor.fontSize or editor.tabSize to a string or something else silly)
5. Observe that a warning is logged and the value does not take effect.
6. Check that setting the preferences whose interface values have been changed reflects the values correctly
> E.g. that the options for `editor.unicodeHighlight.includeComments` really are `true`, `false` (booleans, not strings) and `'inUntrustedWorkspace'`

#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
